File: app/system/modules/MoodleXmlConverterModule.java
package system.modules;

import com.google.inject.AbstractModule;
import util.xml.MoodleXmlExporter;
import util.xml.MoodleXmlExporterImpl;
import util.xml.MoodleXmlImporter;
import util.xml.MoodleXmlImporterImpl;

public class MoodleXmlConverterModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(MoodleXmlExporter.class).to(MoodleXmlExporterImpl.class);
        bind(MoodleXmlImporter.class).to(MoodleXmlImporterImpl.class);
    }
}

File: app/system/modules/MoodleXmlConverterModule.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package system.modules

import com.google.inject.AbstractModule
import services.xml.{MoodleXmlExporter, MoodleXmlExporterImpl, MoodleXmlImporter, MoodleXmlImporterImpl}

class MoodleXmlConverterModule extends AbstractModule:
  override def configure(): Unit =
    bind(classOf[MoodleXmlExporter]).to(classOf[MoodleXmlExporterImpl])
    bind(classOf[MoodleXmlImporter]).to(classOf[MoodleXmlImporterImpl])
